regparser/layer/external_citations.py

```diff
@@ -33,13 +33,19 @@ def reformat_citation(self, citation):
         return [c for c in citation if c not in [
             'U.S.C.', 'CFR', 'part', '.', 'Public', 'Law', '-']]
 
-    def parse(self, text, parts=None):
+    def parse(self, text, parts=None, title=None):
         """ Parse the provided text, pulling out all the citations. """
         parser = grammar.regtext_external_citation
 
         cm = defaultdict(list)
         citation_strings = {}
         for citation, start, end in parser.scanString(text):
+            # Discard citations of form XX CFR YY if XX and YY are the title
+            # and part being parsed
+            if (citation[0] == self.cfr_title and
```

Comment:
I think you can do: 

citation == (self.cfr_title, 'CFR', parts[0])

or something similar. That might be a clearer format. 

Also, should this be the test? Or should 'ignore everything that is an internal citation' be the test? I suspect this is easier but wanted to throw that out there.